prediction/src/tests/test_classification.py

```diff
@@ -1,15 +1,15 @@
 from ..algorithms.classify import trained_model
 
 
-# def test_classify_predict_load(metaimage_path, model_path):
-#     assert not trained_model.predict(metaimage_path, [], model_path)
-#
-#
-# def test_classify_dicom(dicom_paths, nodule_locations, model_path):
-#     predicted = trained_model.predict(dicom_paths[0], nodule_locations, model_path)
-#     assert predicted
-#     assert 0 <= predicted[0]['p_concerning'] <= 1
-#
+def test_classify_predict_load(metaimage_path, model_path):
+    assert not trained_model.predict(metaimage_path, [], model_path)
+
+
+def test_classify_dicom(dicom_paths, nodule_locations, model_path):
+    predicted = trained_model.predict(dicom_paths[0], nodule_locations, model_path)
+    assert predicted
+    assert 0 <= predicted[0]['p_concerning'] <= 1
+
 
```

Comment:
I'm wondering if this will pass travis or hit the 'memory' error issue... I think we may need to move this hunk over to #307.

Reply:
According to the current [Travis log](https://travis-ci.org/concept-to-clinic/concept-to-clinic/builds/335762039#L3322-L3334) and [previous one](https://travis-ci.org/concept-to-clinic/concept-to-clinic/builds/333643030#L3318-L3330) from PR #298 whit commented test, classification model is not responsible for the `memory errosr`s.

Reply:
hmmm, looks like they all failed with a `memory error`...

Reply:
But it worth to move [`Makes relevant scale factor for simple_3d_model`](https://github.com/concept-to-clinic/concept-to-clinic/pull/306/commits/ad5677a8e7f2caa3f4b16f838e40b1bd964cd587) over to #307

Reply:
FWIW, the latest commit also failed as well... so at least we know the error is due to these PRs

Reply:
Though it goes with submitting new DATA_SHAPE. That makes me think that DATA_SHAPE was the reason for `memory error`s, isn't it?

Reply:
The last commit is #298.. and the [log is here](https://travis-ci.org/concept-to-clinic/concept-to-clinic/builds/333643030) same error.

Reply:
Though it works [here](https://travis-ci.org/concept-to-clinic/concept-to-clinic/builds/333559037#L3315-L3326) in PR #307

Reply:
[This](https://travis-ci.org/concept-to-clinic/concept-to-clinic/builds/335784803#L3324) error occurred due to re-estimation of coordinates and fixed in PR #308

Reply:
@reubano, thank you for the remarks, I've checked locally and turns out that the issue was not only in DATA_SHAPE but also in [memory leak when using tensorflow](https://github.com/keras-team/keras/issues/2102), thus I've added `clear_session()` as it was suggested and it corrects the situation for me.

Reply:
I also include an ability to [clear model along with `pull_ct` and `pull_patches`](https://github.com/vessemer/concept-to-clinic/blob/2df7b6c0120d1798cd6331547bb46637bdeb736f/prediction/src/algorithms/classify/src/lr3dcnn/model.py#L312-L316)  which may be useful if [`feed`](https://github.com/vessemer/concept-to-clinic/blob/2df7b6c0120d1798cd6331547bb46637bdeb736f/prediction/src/algorithms/classify/src/lr3dcnn/model.py#L151) will be called more than once for the same [Model](https://github.com/vessemer/concept-to-clinic/blob/2df7b6c0120d1798cd6331547bb46637bdeb736f/prediction/src/algorithms/classify/src/lr3dcnn/model.py#L12) instance. Though, those proprieties can't be responsible for the memory errors. I'll rerun locally with `htop` to provide the information of RAM load.

Reply:
![screenshot from 2018-02-01 00-33-26](https://user-images.githubusercontent.com/9470024/35653982-f9e1346a-06ea-11e8-93c4-527c1bf4d61e.png)
![screenshot from 2018-02-01 00-34-55](https://user-images.githubusercontent.com/9470024/35653983-fa0e303c-06ea-11e8-81a8-fc8c3752ba03.png)
![screenshot from 2018-02-01 00-35-40](https://user-images.githubusercontent.com/9470024/35653984-fa33635c-06ea-11e8-9f88-d2c5d683244b.png)
![screenshot from 2018-02-01 00-36-21](https://user-images.githubusercontent.com/9470024/35653985-fa5a5c96-06ea-11e8-932b-da4aeb9d08e6.png)
![screenshot from 2018-02-01 00-38-43](https://user-images.githubusercontent.com/9470024/35653986-fa7a2080-06ea-11e8-8e76-94605b9e06f0.png)
![screenshot from 2018-02-01 00-45-07](https://user-images.githubusercontent.com/9470024/35653987-fa9d96dc-06ea-11e8-9b51-8346ecbaa671.png)
![screenshot from 2018-02-01 00-47-44](https://user-images.githubusercontent.com/9470024/35653989-fb000416-06ea-11e8-8211-d0e294887f19.png)
![screenshot from 2018-02-01 00-52-15](https://user-images.githubusercontent.com/9470024/35653990-fb2656de-06ea-11e8-99f8-4f7fc2903de7.png)

Reply:
As you can see the main impact in `memory error` was made from the segmentation itself.